pkg/security/secl/rules/fim_unix.go

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build unix
+
+// Package rules holds rules related files
+package rules
+
+import (
+	"fmt"
+	"strings"
+)
+
+type expandedRule struct {
+	id   string
+	expr string
+}
+
+func expandFim(baseID, baseExpr string) []expandedRule {
+	if !strings.Contains(baseExpr, "fim.write.file.") {
+		return []expandedRule{
+			{
+				id:   baseID,
+				expr: baseExpr,
+			},
+		}
+	}
+
+	var expandedRules []expandedRule
+	for _, eventType := range []string{"open", "chmod", "chown", "link", "rename", "unlink", "utimes"} {
+		expr := strings.Replace(baseExpr, "fim.write.file.", fmt.Sprintf("%s.file.", eventType), -1)
+		if eventType == "open" {
+			expr = fmt.Sprintf("(%s) && open.flags & (O_CREAT|O_TRUNC|O_APPEND|O_RDWR|O_WRONLY) > 0", expr)
+		}
+
+		id := fmt.Sprintf("__fim_expanded_%s_%s", eventType, baseID)
```

Comment:
I wonder if we should set the "groupID" to the baseID to keep track of it?

Reply:
I think the user-facing ID will still be the baseID because the rest of the code uses the policy-defined ID and not the expanded one, so there's no need to set the groupID to the baseID @paulcacheux let me know if I'm wrong here
Also know that the policy-defined ID and the eval.RuleID can be different what do we think about renaming the `eval.Rule.ID` field to `eval.Rule.ExpandedID` so that we don't mix up the two?

Reply:
yes but I think it would be nice to include the group id in the expanded ID as well for unicity

agreed on `ExpandedID`

Reply:
If it's ok for you I will do the renaming in a follow up PR, otherwise this adds a lot of review noise